vmdb/app/models/mixins/naming_index_mixin.rb

```diff
@@ -0,0 +1,20 @@
+module NamingIndexMixin
+  extend ActiveSupport::Concern
+
+  NAMING = 'naming'
+
+  included do
+    has_many :naming_indexes, :as => :resource, :dependent => :destroy, :class_name => "CustomAttribute", :conditions => {:source => NAMING}
+  end
+
+  def next_naming_index(name)
+    with_lock do
```

Comment:
I think this is not the right kind of lock...I think you want `MiqRegion.my_region.lock`